packages/network/src/v2/schemas/decodeData.ts

```diff
@@ -29,13 +29,20 @@ export const decodeData = (schema: TableSchema, hexData: string): Record<number,
   }
 
   if (schema.dynamicFields.length > 0) {
-    const dynamicDataLayout = new DataView(bytes.slice(schema.staticDataLength, schema.staticDataLength + 32).buffer);
+    const dynamicDataLayout = bytes.slice(schema.staticDataLength, schema.staticDataLength + 32);
     bytesOffset += 32;
 
-    const dynamicDataLength = dynamicDataLayout.getUint32(0);
+    // keep in sync with PackedCounter.sol
+    const packedCounterAccumulatorType = SchemaType.UINT56;
+    const packedCounterCounterType = SchemaType.UINT40;
+    const dynamicDataLength = decodeStaticField(packedCounterAccumulatorType, dynamicDataLayout, 0) as number;
 
     schema.dynamicFields.forEach((fieldType, i) => {
-      const dataLength = dynamicDataLayout.getUint16(4 + i * 2);
+      const dataLength = decodeStaticField(
+        packedCounterCounterType,
+        dynamicDataLayout,
+        getStaticByteLength(packedCounterAccumulatorType) + i * getStaticByteLength(packedCounterCounterType)
+      ) as number;
```

Comment:
haven't actually tested this yet! please test before merging!

I assume that before these client changes, array values wouldn't decode properly (they'd be sliced in the wrong places). These changes should address that.